tests/pthread/test_pthread_preallocates_workers.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <pthread.h>
+#include <sys/types.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <unistd.h>
+#include <errno.h>
+#include <emscripten.h>
+#include <emscripten/threading.h>
+#include <vector>
+
+static void *thread_start(void *arg)
+{
+  // This should be long enough for threads to pile up.
+  sleep(2);
```

Comment:
I think this is at risk of being a flakey test. Can we loop with a sleep until some condition is true?

Reply:
Yes - will ping with a fix.

Reply:
Done.